eval.py

```diff
@@ -37,7 +37,7 @@ def init_model(self):
         self.session_opti = onnxruntime.SessionOptions()
         self.session_opti.enable_mem_pattern = False
         self.provider = 'CPUExecutionProvider' # or DmlExecutionProvider
-        self.session = onnxruntime.InferenceSession(self.model_path, self.session_opti)
+        self.session = onnxruntime.InferenceSession(self.model_path, self.session_opti, providers=["CPUExecutionProvider"])

```

Comment:
请问为何引入此处修改？当前 39/41 行的 providers 设置方法在当前 onnxruntime 版本（1.13.1）下并未废弃。

Reply:
um 我有收到 Error 耶

```
ValueError: This ORT build has ['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'], ...)
Use case
onnxruntime-gpu 1.10 requires providers

elif onnx:  # ONNX Runtime
    LOGGER.info(f'Loading {w} for ONNX Runtime inference...')
    check_requirements(('onnx', 'onnxruntime-gpu' if torch.cuda.is_available() else 'onnxruntime'))
    import onnxruntime
    if torch.cuda.is_available():
         session = onnxruntime.InferenceSession(w, None, providers=["CUDAExecutionProvider"])
    else:
          session = onnxruntime.InferenceSession(w, None)
Additional
No response

Are you willing to submit a PR?
 Yes I'd like to help by submitting a PR!
@[guishilike](https://github.com/guishilike) guishilike added the [enhancement](https://github.com/ultralytics/yolov5/labels/enhancement) New feature or requestlabel [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#event-5733388772)
@[github-actions](https://github.com/apps/github-actions)
github-actions bot commented [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#issuecomment-988751254)
👋 Hello @guishilike, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://github.com/ultralytics/yolov5/wiki#tutorials) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data) all the way to advanced concepts like https://github.com/ultralytics/yolov5/issues/607.

If this is a 🐛 Bug Report, please provide screenshots and minimum viable code to reproduce your issue, otherwise we can not help you.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset images, training logs, screenshots, and a public link to online [W&B logging](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data#visualize) if available.

For business inquiries or professional support requests please visit https://ultralytics.com/ or email Glenn Jocher at [glenn.jocher@ultralytics.com](mailto:glenn.jocher@ultralytics.com).

Requirements
[Python>=3.6.0](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [PyTorch>=1.7](https://pytorch.org/get-started/locally/). To get started:

$ git clone https://github.com/ultralytics/yolov5
$ cd yolov5
$ pip install -r requirements.txt
Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

Google Colab and Kaggle notebooks with free GPU: Open In Colab Open In Kaggle
Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/GCP-Quickstart)
Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/AWS-Quickstart)
Docker Image. See [Docker Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart) Docker Pulls
Status
CI CPU testing

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

@glenn-jocher
Member
[glenn-jocher](https://github.com/glenn-jocher) commented [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#issuecomment-988759587)
@guishilike thanks for the bug report! I'm able to reproduce. I will work on a fix for this.

@[glenn-jocher](https://github.com/glenn-jocher) glenn-jocher mentioned this issue [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#ref-pullrequest-1074349666)
https://github.com/ultralytics/yolov5/pull/5918
 Merged
[glenn-jocher](https://github.com/glenn-jocher) added a commit that referenced this issue [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#ref-commit-d80d80a)
@glenn-jocher
[Add ONNX inference providers](https://github.com/ultralytics/yolov5/commit/d80d80a8287df7e5d884c482e6d38297043bae57) …
[d80d80a](https://github.com/ultralytics/yolov5/commit/d80d80a8287df7e5d884c482e6d38297043bae57)
@[glenn-jocher](https://github.com/glenn-jocher) glenn-jocher linked a pull request [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#event-5733487874) that will close this issue
https://github.com/ultralytics/yolov5/pull/5918
 Merged
@[glenn-jocher](https://github.com/glenn-jocher) glenn-jocher closed this as [completed](https://github.com/ultralytics/yolov5/issues?q=is%3Aissue+is%3Aclosed+archived%3Afalse+reason%3Acompleted) in https://github.com/ultralytics/yolov5/pull/5918[on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#event-5733576422)
[glenn-jocher](https://github.com/glenn-jocher) added a commit that referenced this issue [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#ref-commit-581dc30)
@glenn-jocher
[Add ONNX inference providers (](https://github.com/ultralytics/yolov5/commit/581dc301a70bef6d3e768adfe1a87b85e50e6268)https://github.com/ultralytics/yolov5/pull/5918[)](https://github.com/ultralytics/yolov5/commit/581dc301a70bef6d3e768adfe1a87b85e50e6268) …
[581dc30](https://github.com/ultralytics/yolov5/commit/581dc301a70bef6d3e768adfe1a87b85e50e6268)
@glenn-jocher
Member
[glenn-jocher](https://github.com/glenn-jocher) commented [on Dec 8, 2021](https://github.com/ultralytics/yolov5/issues/5916#issuecomment-988775262)
@guishilike good news 😃! Your original issue may now be fixed ✅ in PR https://github.com/ultralytics/yolov5/pull/5918. To receive this update:

[Git](https://github.com/ultralytics/yolov5) – git pull from within your yolov5/ directory or git clone https://github.com/ultralytics/yolov5 again
[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/) – Force-reload model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)
[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) – View updated notebooks Open In Colab Open In Kaggle
[Docker](https://hub.docker.com/r/ultralytics/yolov5) – sudo docker pull ultralytics/yolov5:latest to update your image Docker Pulls
Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other issues you discover or feature requests that come to mind. Happy trainings with YOLOv5 🚀!

[bfineran](https://github.com/bfineran) added a commit to neuralmagic/yolov5 that referenced this issue [on Apr 9](https://github.com/ultralytics/yolov5/issues/5916#ref-commit-14eb1dc)
@bfineran
@KSGulin
[Update Neuralmagic-YoloV5 integration to v6.1 (](https://github.com/neuralmagic/yolov5/commit/14eb1dc8980d06917bd5113e0183e97352c880f5)https://github.com/neuralmagic/yolov5/pull/29[)](https://github.com/neuralmagic/yolov5/commit/14eb1dc8980d06917bd5113e0183e97352c880f5) …
[14eb1dc](https://github.com/neuralmagic/yolov5/commit/14eb1dc8980d06917bd5113e0183e97352c880f5)
[KSGulin](https://github.com/KSGulin) added a commit to neuralmagic/yolov5 that referenced this issue [on Apr 14](https://github.com/ultralytics/yolov5/issues/5916#ref-commit-a621b7c)
@bfineran
@KSGulin
[Update Neuralmagic-YoloV5 integration to v6.1 (](https://github.com/neuralmagic/yolov5/commit/a621b7c79f352b87fbf8fa8aa4c8e29540d25ece)https://github.com/neuralmagic/yolov5/pull/29[)](https://github.com/neuralmagic/yolov5/commit/a621b7c79f352b87fbf8fa8aa4c8e29540d25ece) …
[a621b7c](https://github.com/neuralmagic/yolov5/commit/a621b7c79f352b87fbf8fa8aa4c8e29540d25ece)
[KSGulin](https://github.com/KSGulin) added a commit to neuralmagic/yolov5 that referenced this issue [on Apr 14](https://github.com/ultralytics/yolov5/issues/5916#ref-commit-2d3963a)
@bfineran
@KSGulin
[Update Neuralmagic-YoloV5 integration to v6.1 (](https://github.com/neuralmagic/yolov5/commit/2d3963a09fe3753686876d2bf7146b5f58915a21)https://github.com/neuralmagic/yolov5/pull/29[)](https://github.com/neuralmagic/yolov5/commit/2d3963a09fe3753686876d2bf7146b5f58915a21) …
[2d3963a](https://github.com/neuralmagic/yolov5/commit/2d3963a09fe3753686876d2bf7146b5f58915a21)
[BjarneKuehl](https://github.com/BjarneKuehl) pushed a commit to fhkiel-mlaip/yolov5 that referenced this issue [on Aug 26](https://github.com/ultralytics/yolov5/issues/5916#ref-commit-b5f5fcb)
@glenn-jocher
[Add ONNX inference providers (](https://github.com/fhkiel-mlaip/yolov5/commit/b5f5fcb29c4394c584ec93ce372ec94304128c2c)https://github.com/ultralytics/yolov5/pull/5918[)](https://github.com/fhkiel-mlaip/yolov5/commit/b5f5fcb29c4394c584ec93ce372ec94304128c2c) …
[b5f5fcb](https://github.com/fhkiel-mlaip/yolov5/commit/b5f5fcb29c4394c584ec93ce372ec94304128c2c)
@[somuchtolearn007](https://github.com/somuchtolearn007)
somuchtolearn007 commented [26 days ago](https://github.com/ultralytics/yolov5/issues/5916#issuecomment-1310445195)
@guishilike good news 😃! Your original issue may now be fixed ✅ in PR https://github.com/ultralytics/yolov5/pull/5918. To receive this update:

* **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again

* **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`

* **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/ultralytics/yolov5)

* **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image [![Docker Pulls](https://camo.githubusercontent.com/280faedaf431e4c0c24fdb30ec00a66d627404e5c4c498210d3f014dd58c2c7e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f796f6c6f76353f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/yolov5)
Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other issues you discover or feature requests that come to mind. Happy trainings with YOLOv5 🚀!

I am facing the same issue
This ORT build has ['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['TensorrtExecutionProvider', 'CUDAExecutionProvider', 'CPUExecutionProvider'], ...)
while I am trying to run SimSwap. How to fix it?

@glenn-jocher
Member
[glenn-jocher](https://github.com/glenn-jocher) commented [21 days ago](https://github.com/ultralytics/yolov5/issues/5916#issuecomment-1315464583)
@somuchtolearn007 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

How to create a Minimal, Reproducible Example
When asking a question, people will be better able to provide help if you provide code that they can easily understand and use to reproduce the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

✅ Minimal – Use as little code as possible to produce the problem
✅ Complete – Provide all parts someone else needs to reproduce the problem
✅ Reproducible – Test the code you're about to provide to make sure it reproduces the problem
For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

✅ Current – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary git pull or git clone a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
✅ Unmodified – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.
If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 Bug Report [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem.

Thank you! 😃

@lekoOwO

 
Add heading textAdd bold text, <Ctrl+b>Add italic text, <Ctrl+i>
Add a quote, <Ctrl+Shift+.>Add code, <Ctrl+e>Add a link, <Ctrl+k>
Add a bulleted list, <Ctrl+Shift+8>Add a numbered list, <Ctrl+Shift+7>Add a task list, <Ctrl+Shift+l>
Directly mention a user or team
Reference an issue, pull request, or discussion
Add saved reply
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Styling with Markdown is supported
Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/ultralytics/yolov5/blob/f8539a680041a9f4fbcc4fcdd8f540724da453af/CONTRIBUTING.md), [security policy](https://github.com/ultralytics/yolov5/blob/f8539a680041a9f4fbcc4fcdd8f540724da453af/.github/SECURITY.md), and [code of conduct](https://github.com/ultralytics/yolov5/blob/f8539a680041a9f4fbcc4fcdd8f540724da453af/.github/CODE_OF_CONDUCT.md).
Assignees
No one assigned
Labels
[enhancement](https://github.com/ultralytics/yolov5/labels/enhancement)
New feature or request
Projects
None yet
Milestone
```

Reply:
了解了